.github/workflows/images.yaml

```diff
@@ -0,0 +1,67 @@
+name: Images
+on:
+  push:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+      - v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+
+
+jobs:
+  build-and-push:
+    if: github.repository == 'cilium/cilium'
+    name: Build and push all images
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: docker://docker.io/cilium/image-maker:bc81755ec8f6c5afcb10a416cef73f99a35fee2c
+        name: Register binfmt from multi-platform builds
+        with:
+          entrypoint: docker
+          args: run --privileged linuxkit/binfmt:5d33e7346e79f9c13a73c6952669e47a53b063d4 
+      - uses: docker://docker.io/cilium/image-maker:bc81755ec8f6c5afcb10a416cef73f99a35fee2c
+        name: Run make lint
+        with:
+          entrypoint: make
+          args: -C images lint
+      - uses: docker://docker.io/cilium/image-maker:bc81755ec8f6c5afcb10a416cef73f99a35fee2c
+        name: Run make runtime-image
+        env:
+          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+        with:
+          entrypoint: make
+          args: -C images runtime-image PUSH=true
+      - uses: docker://docker.io/cilium/image-maker:bc81755ec8f6c5afcb10a416cef73f99a35fee2c
+        name: Run make builder-image
+        env:
+          DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+        with:
+          entrypoint: make
+          args: -C images builder-image PUSH=true
```

Comment:
Why do we need to build the runtime and builder images each time? We shouldn't build those as part of the release process because we need to build these first, test them against the CI and only them publish the Cilium images based on these.

Perhaps we should have another GH action that will build these images independently?

This comment can be addressed as a follow up.

Reply:
Actually, these won't be rebuilt unless there were changes. runtime and builder are scoped to their subdirs, and we use git tree hashes as tags. The logic is implemented in [`image/scripts/build-image.sh`](https://github.com/cilium/cilium/blob/ea59fd095d7c38f70a82a9498219f69f1f6a192c/images/scripts/build-image.sh#L72-L92). So with runtime and builder the tag changes only when there are actual changes in the directory.

Reply:
Please add a comment here about that behavior.

Reply:
Sure, it's not in any explicit right now. There is a mention in the readme, however.